plugins/Kaleidoscope-LongPress/README.md

```diff
@@ -0,0 +1,125 @@
+# LongPress
+
+LongPress allows you to type different characters when long-pressing a key
+rather than tapping it.
+
+It is derived from the AutoShift plugin and integrates its functionality. When
+using LongPress it supersedes the AutoShift plugin. They conflict with each
+other and should not be used together.
+
+
+## Setup
+
+To use the plugin put the following into your .ino file:
+
+```c++
+#include <Kaleidoscope-LongPress.h>
+
+KALEIDOSCOPE_INIT_PLUGINS(LongPress);
+```
+
+## Configuration
+
+To do anything useful some configuration is necessary.
+
+### Long-press mappings
+
+To define the keys that should behave differently on long-press use include a definition like the following:
+
+```c++
+LONGPRESS(
```

Comment:
Perhaps a naive question - how does the configuration by KeyAddr interact with multiple layers on the keyboard? Is long press behavior scoped to a single layer or does it override the behavior no matter what layer someone is on?
Ideally, long-press behavior would be configurable by layer, I think.